bundles/specmate-migration/src/com/specmate/migration/internal/services/Migrator20181108.java

```diff
@@ -0,0 +1,74 @@
+package com.specmate.migration.internal.services;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+import com.specmate.common.SpecmateException;
+import com.specmate.config.api.IConfigService;
+import com.specmate.connectors.api.IProjectConfigService;
+import com.specmate.dbprovider.api.IDBProvider;
+import com.specmate.dbprovider.api.migration.IAttributeToSQLMapper;
+import com.specmate.migration.api.IMigrator;
+
+@Component(property = "sourceVersion=20181108", service = IMigrator.class)
+public class Migrator20181108 implements IMigrator {
+
+	private IDBProvider dbProvider;
+	private IConfigService configService;
+
+	@Override
+	public String getSourceVersion() {
+		return "20181108";
+	}
+
+	@Override
+	public String getTargetVersion() {
+		return "20181210";
+	}
+
+	@Override
+	public void migrate(Connection connection) throws SpecmateException {
+		IAttributeToSQLMapper aMapper = dbProvider.getAttributeToSQLMapper("model/base", getSourceVersion(),
+				getTargetVersion());
+		aMapper.migrateNewBooleanAttribute("Folder", "library", false);
+
+		String[] projectsNames = configService.getConfigurationPropertyArray(IProjectConfigService.KEY_PROJECT_NAMES);
+
+		try {
+			if (projectsNames != null) {
+				for (int i = 0; i < projectsNames.length; i++) {
+					String projectName = projectsNames[i];
+					String projectLibraryKey = IProjectConfigService.PROJECT_PREFIX + projectName
+							+ IProjectConfigService.KEY_PROJECT_LIBRARY;
+					String[] libraryFolders = configService.getConfigurationPropertyArray(projectLibraryKey);
+					if (libraryFolders != null) {
+						for (int j = 0; j < libraryFolders.length; j++) {
+							String libraryFolder = libraryFolders[j];
+							String sql = "UPDATE FOLDER set library = true where id = '" + libraryFolder + "'";
```

Comment:
Können wir sicher sein, dass die id eindeutig ist? Vermutlich nicht.
Können wir zumindest noch checken, dass der aktuelle Parent-Ordner ein (bzw. der richtige) Projektordner ist?

Reply:
The migrator ensures now that only those folders are marked as library folders that:
1) are, at the time of migration, configured in a project as library folders
2) are indeed child folders of the configured project with library folders 

1) is a necessary constraint (i.e. we cannot mark as library folders all folders that are sub-folders of projects, in general, but only those that are configured as library folders). This needs to be considered when running the migration on a production system, e.g. by temporarily adding all removed library folders again to the configuration.